### ISSUE TITLE
mermaid charts configuration

### ISSUE THREAD

**Issue by gd047** [rich-iannone/DiagrammeR#98]
Is it possible to configure mermaid charts, i.e. using mermaid.ganttConfig, as shown here? http://knsv.github.io/mermaid/gantt.html#configuration

I would like to be able to change the default x axis labels ("w.01", "w.02") in gantt charts, since there is no information in the default gantt chart about the dates the displayed activites occur.

P.S. it also seems to me rather difficult to automatically create this string that is passed to mermaid, using data stored in a dataframe, so I think it would be a good idea to provide an easier way to do that.

**Comment by timelyportfolio** [comment]
See if #77 helps.  I will revisit to see if there is a better way or updates on the mermaid side.

**Comment by rich-iannone** [comment]
Thanks @timelyportfolio, that does help. I do think a method to exploit data residing in a data frame would be the way to go. I'll devote at a bit of time toward getting this implemented! Then some attributes (like the axis type) passed in before creating the Gantt chart would be ideal.

**Comment by Gelu74** [comment]
any updates to this? I am new to mermaid and I am trying to use it to create a gantt chart in R, I would like the axis to be in months rather than weeks, any pointers on how to achieve this would be more than welcomed.

**Comment by timelyportfolio** [comment]
@Gelu74, is #77 helpful or completely incomprehensible?  I have not had a chance to work on making it smooth and painless.

**Comment by Gelu74** [comment]
Hi @timelyportfolio, not completely incomprehensible but not quite what I wanted, that code worked but it did just change the label from a week number into a MonthDayYear, I would like the tickmarks to appear only every month. In that code you "filter for Mondays or day of week = 1" is there any way to filter for first day of the month or, preferably, 15th of each month so the label is centered in each month?
Thanks

**Comment by benmarwick** [comment]
Agreed, a bit more flexibility with the time axis tick marks and labels would make this approach to Gnatt charting much more useful. Controlling the tick and label spacing would be really great feature.
